lib/test/test_tls.c

```diff
@@ -833,6 +835,46 @@ static void test_set_mem(void *z)
 end:    ;
 }
 
+static void test_cipher_tlsv12(void *z)
```

Comment:
I think it would make sense to also have a Python based test for this functionality. To make sure the pgbouncer integration with this new API works as expected.